### DISCUSSION TITLE
Feeding the Select Form dynamically

### DISCUSSION THREAD

**Discussion by Muratcol** [orchidsoftware/platform#1538] [Q&A]
Hello guys!

I want to send emails to our customers by using Orchid at the company I work for. However, we classified these customers according to their level of authority. And we have to send these emails according to their authorization levels.

I can now select the customer information we got from the database with the select form according to the authorization level, but this is not dynamic. I can only take one group at a time.

Currently, It's working like this:
```php                
Select::make('mail.clients.')
          ->title("Recipients")
          ->multiple()
          ->placeholder('Email addresses')
          ->help('Enter the campaign designer accounts that you would like to send this mail to.')
          ->fromQuery(Client::where('type', '=', '1'), 'email');

```

`Client::where('type', '=', '1')`  I want to make this '1' to dynamic. When I select type 2 for example, this part should looks like this `Client::where('type', '=', '2')` 
How can I do this?

**Comment by tabuna** [comment]
Hi, do you want to get the value `2` from the screen query?

If it is on the screen, then you can use properties
```php
/**
 * Query data.
 *
 * @return array
 */
public function query(): array
{
    $this->userType = 2;
    
    return [];
}


/**
 * Views.
 *
 * @return Layout[]
 */
public function layout(): array
{
    Layout::rows([
        Select::make('mail.clients.')
            ->title("Recipients")
            ->multiple()
            ->placeholder('Email addresses')
            ->help('Enter the campaign designer accounts that you would like to send this mail to.')
            ->fromQuery(Client::where('type', '=', $this->userType), 'email'),
    ]);
}
```


If you are using a separate class, then don't worry. You can also get all the values of the screen query:

```php

// screen
public function query(): array
{
    return [
        'userType' => 2
    ];
}

// Row layout class
class ClientRow extends Rows
{
    /**
     * Get the fields elements to be displayed.
     *
     * @return Field[]
     */
    protected function fields(): array
    {
        return [
            Select::make('mail.clients.')
                ->title("Recipients")
                ->multiple()
                ->placeholder('Email addresses')
                ->help('Enter the campaign designer accounts that you would like to send this mail to.')
                ->fromQuery(Client::where('type', '=', $this->query->get('userType')), 'email'),
        ];
    }
}

```

**Reply by Muratcol** [reply]
Actually, I was thinking of doing all the work here and not using the query, but it is a bit difficult, I think. In the database, customer information in the database look like this: 

name: John Doe level: superadmin, 
name: John Doe2 level: superadmin, 
name: Jane Doe level: user, 
name: Jane Doe2 level: user, 
name: Mr.Smith level: campaignManager

 I want to send emails group by group, for example the user will select level superadmins with select form and all superadmins will receive the same email.
 
 Thank you for the answer. It can certainly be done as you recommend. But I was wondering if I could do it as I mentioned above?
